src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -333,6 +333,19 @@ public final class Guardrails implements GuardrailsMBean
                                : format("The keyspace %s has a replication factor of %s, below the failure threshold of %s.",
                                         what, value, threshold));
 
+    /**
+     * Guardrail on the maximum replication factor.
+     */
+    public static final MaxThreshold maximumReplicationFactor =
+    new MaxThreshold("maximum_replication_factor",
+                  state -> CONFIG_PROVIDER.getOrCreate(state).getMaximumReplicationFactorWarnThreshold(),
+                  state -> CONFIG_PROVIDER.getOrCreate(state).getMaximumReplicationFactorFailThreshold(),
+                  (isWarning, what, value, threshold) ->
+                  isWarning ? format("The keyspace %s has a replication factor of %s, above the warning threshold of %s.",
```

Comment:
May be able to tidy this up with something like:
```
format("The keyspace %s has a replication factor of %s, above the %s threshold of %s),
             what, value, isWarning ? "warning" : "failure", threshold);
```
Gets you down to one line with the ternary on the string argument only.